datafusion/functions-aggregate-common/src/tdigest.rs

```diff
@@ -641,8 +641,15 @@ impl TDigest {
             v => panic!("invalid centroids type {v:?}"),
         };
 
-        let max = cast_scalar_f64!(&state[3]);
-        let min = cast_scalar_f64!(&state[4]);
+        let mut max = cast_scalar_f64!(&state[3]);
+        let mut min = cast_scalar_f64!(&state[4]);
+
+        if !min.is_finite() {
+            min = f64::NEG_INFINITY;
+        }
+        if !max.is_finite() {
+            max = f64::INFINITY;
+        }
 
         assert!(max.total_cmp(&min).is_ge());
```

Comment:
I think another fix would be to add a guard for this assertion.
```rust
if min.is_finite() && max.is_finite() {
    assert!(max.total_cmp(&min).is_ge());
}
```

The current change causes the following two queries to have different signs. Even though I think their results should be NaN, it's better to keep them consistent.
```sh
> select approx_percentile_cont_with_weight('NaN'::Double, 0, 0);
+-------------------------------------------------------------------+
| approx_percentile_cont_with_weight(Utf8("NaN"),Int64(0),Int64(0)) |
+-------------------------------------------------------------------+
| inf                                                               |
+-------------------------------------------------------------------+
1 row(s) fetched.
Elapsed 0.006 seconds.

> SELECT approx_percentile_cont_with_weight('NaN'::DOUBLE, 0, 0) FROM t1 WHERE t1.v1;
+-------------------------------------------------------------------+
| approx_percentile_cont_with_weight(Utf8("NaN"),Int64(0),Int64(0)) |
+-------------------------------------------------------------------+
| -inf                                                              |
+-------------------------------------------------------------------+
```

Reply:
@jonahgao Good catch!